src/cborvalidation.c

```diff
@@ -436,16 +436,15 @@ static CborError validate_container(CborValue *it, int containerType, int flags,
 {
     CborError err;
     const uint8_t *previous = NULL;
-    const uint8_t *previous_end;
+    const uint8_t *previous_end = NULL;
```

Comment:
This silences the warning, but does it really fix the problem?  This is essentially transforming the `size_t bytelen1 = (size_t)(previous_end - previous)` statement into `size_t bytelen1 = (size_t)(0 - previous)` in the case where previous_end wouldn't be initialized.

Reply:
There was no problem to begin with. Both uses of those variables happen after
```c++
    if (containerType != CborMapType)
        continue;
```
[line 467]

That means `containerType` was `CborMapType`, which means `current` **is** properly intialized. In turn, that means `previous` will be properly initialized for the next iteration.